vtr_flow/scripts/python_libs/vtr/util.py

```diff
@@ -0,0 +1,515 @@
+            raise
+
+def print_verbose(min_verbosity, curr_verbosity, string, endl=True):
+    """
+    Print string if curr_verbosity is gteq min_verbosity
+    """
+    if curr_verbosity >= min_verbosity:
+        if endl:
+            print(string)
+        else:
+            print (string,)
+
+def find_vtr_file(filename, is_executable=False):
+    """
+    Attempts to find a VTR related file by searching the environment.
+    
+    Checking the following places:
+        1) System path (if is_executable=True)
+        2) The inferred vtr root from environment variables or the script file location
+
+    """
+    #We assume exectuables are specified in the unix style (no .exe),
+    # if it was specified with .exe, strip it off
+    file_path = PurePath(filename)
+    if file_path.suffix == ".exe":
+        filename = file_path.name
+
+    #
+    #Check if it is on the path (provided it is executable)
+    #
+    if is_executable:
+        #Search first for the non-exe version
+        result = distutils_spawn.find_executable(filename)
+        if result:
+            return result
+
+        #If not found try the .exe version
+        result = distutils_spawn.find_executable(filename + '.exe')
+        if result:
+            return result
+
+    vtr_root = find_vtr_root()
+
+    #Check the inferred VTR root
+    result = find_file_from_vtr_root(filename, vtr_root, is_executable=is_executable)
+    if result:
+        return result
+
+    #Since we stripped off the .exe, try looking for the .exe version as a last resort (i.e. on Windows/cygwin)
+    if is_executable:
+        result = find_file_from_vtr_root(filename + '.exe', vtr_root, is_executable=is_executable)
+        if result:
+            return result
+
+    raise ValueError("Could not find {type} {file}".format(type="executable" if is_executable else "file",file=filename))
+
+def find_file_from_vtr_root(filename, vtr_root, is_executable=False):
+    """
+    Given a vtr_root and a filename searches for the file recursively under some common VTR directories
+    """
+    for subdir in ['vpr', 'abc', 'abc_with_bb_support', 'ODIN_II', 'vtr_flow', 'ace2']:
+        directory_path = (Path(vtr_root) / subdir)
+        for file_path in directory_path.glob('**/*'):
+            if file_path.name == filename:
+                if file_path.is_file():
+                    if is_executable:
+                        #Found an executable file as required
+                        if os.access(str(file_path), os.X_OK):
+                            return str(file_path)
+                    else:
+                        #Found a file as required
+                        return str(file_path)
+    return None
+
+def find_vtr_root():
+    for env_var in ['VTR_ROOT']:
+        if env_var in os.environ:
+            return os.environ[env_var]
+
+    #We assume that this file is in <vtr_root>/vtr_flow/python_libs/verilogtorouting
+    inferred_vtr_root = Path(__file__).parent / '../../../..'
```

Comment:
Use .parent.parent.parent.parent instead of '../../../..'

Reply:
This has been fied